### PR TITLE
FE- Save Heat Times (First Time)

### PR DESCRIPTION
**Pull request by viriponce**
This PR address issue #150 which is related to issue #149.

When clicking "Save" action, all the heat times on the lane should be register. The edit action on the main table should not be visible any more and for each row on the sub table a new edit button should appear.

### Implementation

1. **_frontend/src/Heat/DetailsByLane.jsx_**

  - Added a new prop `onLaneDataUpdate` to handle updating the heat times displayed on the tables after successfully saving them.
  - Call the API to save the heat times after clicking the save button.
  - In case of an error calling the API, an alert is displayed to notify the user.
  - Add `actionsLaneSubTable` with an Edit Action to be visible in a row when there is a registered heat time.

2. **_frontend/src/components/Common/ExpandableTable.jsx_**

  - Added a new prop `subTableActions` to allow the subtable include actions. 
  - The icons for the actions defined in the `subTableActions` are rendered.

3. **_frontend/src/Event/EventDetails.jsx_**

  - Implement `handleLaneDataUpdate`  function to handle updating the heat_time on `laneData` and `heatData` upon saving them in order to display them on the tables by lane and by heat . 
  - This function is passed as a value for prop `onLaneDataUpdate`.

### UI Preview

- Edit action in the main table is visible for lanes without heat times. 
- Edit action in the sub table is visible for the rows with heat times.

Screenshot 2024-12-01 at 11 31 33 PM

- Clicking edit on a Lane render a HeatTimeForm on rows with athletes. The user registers heat time for the athletes.
Screenshot 2024-12-01 at 11 32 06 PM

-  Clicking save registers the heat times and are now displayed on the table. The edit option is now displayed for these rows since there is a heat time registered for each one.
Screenshot 2024-12-01 at 11 32 24 PM

- The heat time is also displayed on the by heat table.
Screenshot 2024-12-01 at 11 32 45 PM

### CODE REVIEW

File: frontend/src/Event/EventDetails.jsx [hunk 1]
```diff
@@ -64,6 +64,32 @@ const EventDetails = ({
     };
   }, [eventId]);
 
+  const handleLaneDataUpdate = (updatedData) => {
```
Comment (MarisiaS) [inline frontend/src/Event/EventDetails.jsx:67]: I am not sure about this function.
The function updates `laneData` and `heatData` locally, without syncing or validating these changes with the backend. If another user makes changes, those updates will not automatically reflect.

**Review by MarisiaS — CHANGES_REQUESTED** [review]
Could you please change the width of .heatTimeFormBox to 48% in App.css?

File: frontend/src/Event/EventDetails.jsx (hunk 1 above)
Comment (viriponce) [reply frontend/src/Event/EventDetails.jsx]: Instead, added a state variable and add it to the useEffect dependency array to trigger the api call to reload the heat and lane details data.

**Review by MarisiaS — APPROVED** [review]
LGTM!
I was able to enter heat times for an event without a problem. I also was able to pop the error Alertbox.

Thanks Viri!

**Pull request merged**